File: 058_length_of_last_word.java
class Solution {
    public int lengthOfLastWord(String s) {
        if (s == null || s.length() == 0) return 0;
        int start = s.length() - 1;
        while (start >= 0 && s.charAt(start) == ' ') {
            start--;
        }
        int length = 0;
        for (int i = start; i >= 0; i--) {
            if (s.charAt(i) == ' ') break;
            else length++;
        }
        return length;
    }
}

File: python/058_length_of_last_word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        last = len(s)
        while last > 0 and s[last - 1] == ' ':
            last = last - 1
        length = 0
        while last > 0 and s[last - 1] != ' ':
            last = last - 1
            length = length + 1
        return length
